test/tests/tsdx-build-default.test.js

```diff
@@ -0,0 +1,75 @@
+const shell = require('shelljs');
+const util = require('../fixtures/util');
+
+shell.config.silent = false;
+
+const fixtureName = 'build-default';
+const stageName = `stage-${fixtureName}`;
+
+describe('tsdx build :: zero-config defaults', () => {
+  beforeAll(() => {
+    util.teardownStage(stageName);
+    util.setupStageWithFixture(stageName, fixtureName);
+  });
+
+  it('should compile files into a dist directory', () => {
+    const output = shell.exec('node ../dist/index.js build');
+
+    expect(shell.test('-f', 'dist/index.js')).toBeTruthy();
+    expect(
+      shell.test('-f', 'dist/build-default.cjs.development.js')
+    ).toBeTruthy();
+    expect(
+      shell.test('-f', 'dist/build-default.cjs.production.min.js')
+    ).toBeTruthy();
+    expect(shell.test('-f', 'dist/build-default.esm.js')).toBeTruthy();
+
+    expect(shell.test('-f', 'dist/index.d.ts')).toBeTruthy();
+
+    expect(output.code).toBe(0);
+  });
+
+  it('should create the library correctly', () => {
```

Comment:
ok, so skipping the `shell.exec` build step is actually problematic -- it means that you can't comment out some of the tests or use `it.only` because order matters now... right, I kind of missed that.

So a better option would be to create a sort of caching function that checks if the same build were just run and skip it if so, and if not, run it. So if things get commented out or `it.only` is used everything still runs properly and order only affects the speed/cache, not the actual output.

Reply:
Now that `util.setupStageWithFixture` is in a `beforeAll` we could move the first `shell.exec` into there too or put all tests that re-use the same `shell.exec` into one nested `describe` with a `beforeAll`, which is probably more correct.

Ideally it should remake the whole fixture and exec per test, but that makes tests much slower, which has been hampering me in dev-test cycles a lot 😕

Reply:
decided to create a naive `execWithCache` function as mentioned above. main problem with `beforeAll` is that `output` must then be a `describe`-scoped variable, which is sub-optimal testing practice